src/odemis/acq/align/spot.py

```diff
@@ -357,31 +358,43 @@ def FindGridSpots(image, repetition):
     Returns
     -------
     spot_coordinates : array like
-        A 2D array of shape (N, 2) containing the coordinates of the spots.
+        A 2D array of shape (N, 2) containing the coordinates of the spots,
+        in respect to the top left of the image.
     translation : tuple of two floats
+        Translation from the origin to the center of the grid in image space,
+        origin is top left of the image. Primary axis points down and the
+        secondary axis points to the right.
     scaling : tuple of two floats
+        Scaling factors for primary and secondary axis.
     rotation : float
+        Rotation in image space, positive rotation is clockwise.
     shear : float
+        Horizontal shear factor. A positive shear factor transforms a coordinate
+        in the positive x direction parallel to the x axis.
 
     """
+    # Find the center coordinates of the spots in the image.
     spot_positions = MaximaFind(image, repetition[0] * repetition[1])
     if len(spot_positions) < repetition[0] * repetition[1]:
         logging.warning('Not enough spots found, returning only the found spots.')
         return spot_positions, None, None, None
-    # Estimate transformation
+    # Estimate the two most common (orthogonal) directions in the grid of spots, defined in the image coordinate system.
     lattice_constants = EstimateLatticeConstant(spot_positions)
+    # The two directions form the transformation matrix. Transpose because we want to describe a grid in the coordinate
+    # system of the lattice, in the image coordinate system.
     transformation_matrix = numpy.transpose(lattice_constants)
```

Comment:
This step is still super funky to me. I will ask Andries if this comment is correct.